### PR TITLE
Create private route

### PR DESCRIPTION
**Pull request by sophiekrag**
Private route is working. Maybe the useContext is an overkill here?

### CODE REVIEW

File: src/utils/PrivateRoute.js [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
 
 function PrivateRoute({ children, ...restOfProps }) {
-  const isAuth = true
-  
-  return isAuth ? 
-    (<Route {...restOfProps}>{children}</Route>)
-       :
-    (<Redirect exact to="/" /> )
+  const { isLoggedIn } = useContext(IsAuthContext);
+  const { isAuth } = useContext(IsAuthContext);
```
Comment (michaelkoek) [inline src/utils/PrivateRoute.js]: Can ben done in one line. You don't have to do import it separately as you're just destructering your `context` which is an object. 

```javascript 
const { isLoggedIn, isAuth } = useContext(IsAuthContext);
```

File: src/utils/PrivateRoute.js [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+
+  isLoggedIn();
```
Comment (michaelkoek) [inline src/utils/PrivateRoute.js]: Wrap a `useEffect` around this function. It's best to not have these "floating" functions hanging around in your component. They should also be called by something. 
In this case a useEffect with an empty dependency array would be good

File: src/utils/PrivateRoute.js [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
+          here
+        </Link>
+      </p>
+    );
+  }
+
+  return isAuth ? (
+    <Route {...restOfProps}>{children}</Route>
+  ) : (
+    <Redirect exact to="/" />
```
Comment (michaelkoek) [inline src/utils/PrivateRoute.js]: This `redirect` will never hit.

You already have this condition `if (!isAuth) {`, which means "if the user is not authenticated than go inside of this `if` block an `return ` the code.

And if the user IS authenticated, that means they can `route` so this code will be executed   `<Route {...restOfProps}>{children}</Route>`


Either have:
```javascript

return  isAuth ?  <Route {...restOfProps}>{children}</Route> : <Redirect exact to="/" />
```
or like this:
```javascript

return isAuth ? (
    <Route {...restOfProps}>{children}</Route>
  ) : (
    <p>
        You need to be logged in. Please login{" "}
        <Link exact to="/">
          here
        </Link>
      </p>
)
```

Comment (michaelkoek) [inline src/utils/PrivateRoute.js]: Shouldn't it redirect to the login page?

**Review by michaelkoek — CHANGES_REQUESTED** [review]

File: src/utils/AuthContext.js [hunk 4]
```diff
@@ -1,15 +1,25 @@
 import React, { useState, createContext } from "react";
+import axiosApi from "./AxiosApi";
 
 export const IsAuthContext = createContext({});
```
Comment (michaelkoek) [inline src/utils/AuthContext.js]: This is a weird variable name. 
IS this the Auth Context? Answer: empty object 🤷 

Only prefix the variable name with `is` if it's a boolean so you can read it like this:
e.g.
```javascript
const age = 12;
const isCorrectAge = age === 12 // which will yield to a "true" or "false" value (boolean)
```

File: src/utils/PrivateRoute.js (hunk 3 above)
Comment (sophiekrag) [reply src/utils/PrivateRoute.js]: Home page is the login page....

File: src/utils/PrivateRoute.js (hunk 2 above)
Comment (sophiekrag) [reply src/utils/PrivateRoute.js]: It's complaining without dependency so now it has one

**Review by michaelkoek — APPROVED** [review]

**Pull request merged**